homeassistant/components/climate/generic_thermostat.py

```diff
@@ -389,14 +577,37 @@ def is_away_mode_on(self):
     def turn_away_mode_on(self):
         """Turn away mode on by setting it on away hold indefinitely."""
         self._is_away = True
-        self._saved_target_temp = self._target_temp
-        self._target_temp = self._away_temp
-        self._async_control_heating()
+        if self._dual_temp:
```

Comment:
I think the whole method can be reduced to (assuming we got rid of `self._away_temp`, as described before):
```
if self._target_temp_low:
  self._saved_target_temp_low = self._target_temp_low
  self._target_temp_low = self._away_temp_heat
if self._target_temp_high:
  self._saved_target_temp_high = self._target_temp_high
  self._target_temp_high = self._away_temp_cool
```

Reply:
I think we can just always set both as the rest of methods will just use the needed one. I've removed the need for `self._dual_temp`